projects/fractional-art.js

```diff
@@ -0,0 +1,199 @@
+    openedVaultsCount += vaults_i.filter(vault => !vault.isClosed).length
+
+    // Append to complete vaults array
+    vaults.push(...vaults_i)
+  }
+  return {vaults, openedVaultsCount}
+}
+
+// Remove vaults tokens (token0) balances as they should not account for TVL
+function clearVaultsTokenBalances(vaults, balances) {
+  const vaults_t0 = vaults.map(v => v.contractAddress.toLowerCase())
+  balances = Object.assign({}, ...
+    Object.entries(balances).filter(([k,v]) => (!vaults_t0.includes(k.toLowerCase()))).map(([k,v]) => ({[k]:v}))
+  )
+  return balances 
+}
+
+// This API returns a list of vaults similar to the following exampleVaultDebug
+function exampleVaultDebug() {
+  return {
+    "symbol":	"DOG",
+    "contractAddress":	"0xbaac2b4491727d78d2b78815144570b9f2fe8899",
+    "pools":	[ {
+      "pool":	"0x7731CA4d00800b6a681D031F565DeB355c5b77dA",
+      "token0":	"0xBAac2B4491727D78D2b78815144570b9f2Fe8899",
+      "token1":	"0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
+      "provider":	"UNISWAP_V3"
+    }],
+    "tokenAddress":	"0xabEFBc9fD2F806065b4f3C237d4b59D9A97Bcac7",
+    "analytics": {
+      "tvlUsd": 21790624
+    },
+    "slug":	"the-doge-nft",
+    "isClosed":	false,
+  }
+}
+
+// Get Fractional.art TVL
+async function tvl(timestamp, block, chainBlocks, chain) {
+  // Get vaults and Compute vaults TVL (trusting fractional rest api)
+  const {vaults, openedVaultsCount} = await retrieveVaultsAPI()
+  const vaulstTVL_api = getVaultsTvlApi(vaults)
+  // note: vault with slug fractional-dream-930 has null analytics and symbol, because it is an ERC1155 not listed on any DEXes
+  
+  // Or try to find all pools associated to vault and account for tokens locked against vault token
+  // In pool: provider, pool, token0, token1
+  const univ2_sushiv1_pools = []
+  const univ3_pools = []
+  vaults.forEach(vault => {
+    vault.pools.forEach(pool => {
+      // Swap token0 and token1 if needed so token0 is always vault token
+      if (vault.contractAddress.toLowerCase() === pool.token1.toLowerCase()) {
+        const tmp = pool.token1
+        pool.token1 = pool.token0
+        pool.token0 = tmp
+      }
+      // Pool provider can be any of ['UNISWAP_V3', 'SUSHISWAP_V1', 'UNISWAP_V2']
+      const provider = pool.provider
+      if ((provider === 'UNISWAP_V2') || (provider === 'SUSHISWAP_V1')) {
+        univ2_sushiv1_pools.push(pool)
+      } else if (provider === 'UNISWAP_V3') {
+        univ3_pools.push(pool)
+      }
+    })
+  })
+  // Concat v2 and v3 pools
+  const v2_v3_pools = univ3_pools.concat(univ2_sushiv1_pools)
+
+  // Retrieve balances from onchain calls
+  let balances = {}
+
+  // Get UNISWAP_V3 LPs
+  // Simply get amount of token0 and token1 allocated to pool contract. And since we only need the token1 it is even more efficient
+  const calls_v3_v2_t0_t1 = v2_v3_pools.map((pool) => ({ 
+      target: pool.token1,
+      params: pool.pool
+    })).concat(v2_v3_pools.map((pool) => ({
```

Comment:
wouldn't it be easier to remove this bit and never call `clearVaultsTokenBalances`?